File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/schema/ChatEvent.java
package ru.iqchannels.sdk.schema;

import androidx.annotation.Nullable;

import java.util.List;

/**
 * Copyright iqstore.ru. All Rights Reserved.
 */
public class ChatEvent {
    public long Id;
    public ChatEventType Type;
    public long ChatId;
    public boolean Public;
    public boolean Transitive;

    @Nullable public Long SessionId;
    @Nullable public Long MessageId;
    @Nullable public Long MemberId;

    public ActorType Actor;
    @Nullable public Long ClientId;
    @Nullable public Long UserId;
    @Nullable public List<ChatMessage> Messages;

    public long CreatedAt;

    // Local
    @Nullable public Client Client;
    @Nullable public User User;
    @Nullable public ChatMessage Message;
    
}

File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/schema/ChatEvent.kt
package ru.iqchannels.sdk.schema

class ChatEvent {
	var Id: Long = 0
	var Type: ChatEventType? = null
	var ChatId: Long = 0
	var Public = false
	var Transitive = false

	var SessionId: Long? = null
	var MessageId: Long? = null
	var MemberId: Long? = null

	var Actor: ActorType? = null
	var ClientId: Long? = null
	var UserId: Long? = null
	var Messages: List<ChatMessage>? = null
	var CreatedAt: Long = 0

	var NextChannelName: String? = null

	// Local
	var Client: Client? = null
	var User: User? = null
	var Message: ChatMessage? = null
}
